src/emqx_session.erl

```diff
@@ -575,19 +576,28 @@ handle_info({dispatch, Topic, Msgs}, State) when is_list(Msgs) ->
 
 %% Dispatch message
 handle_info({dispatch, Topic, Msg = #message{headers = Headers}},
-            State = #state{subscriptions = SubMap, topic_alias_maximum = TopicAliasMaximum}) when is_record(Msg, message) ->
+            State = #state{subscriptions = SubMap,
+                           topic_alias_maximum = TopicAliasMaximum}) when is_record(Msg, message) ->
     TopicAlias = maps:get('Topic-Alias', Headers, undefined),
-    if
-        TopicAlias =:= undefined orelse TopicAlias =< TopicAliasMaximum ->
+    case TopicAlias of
+        undefined ->
```

Comment:
Does `TopicAlias =:= undefined` imply mqtt v3 ?
It would be confusing when we read this code again some time later or for people come across for the first time.
Ideally, the client should subscribe with `nl` set to `true` when `proto_ver` is lower than 5.0.
The least we can do here is to add a comment.